db/snapshot/backup_task.cc

```diff
@@ -99,52 +111,132 @@ future<> backup_task_impl::upload_component(sstring name) {
     }
 }
 
-future<> backup_task_impl::do_backup() {
-    if (!co_await file_exists(_snapshot_dir.native())) {
-        throw std::invalid_argument(fmt::format("snapshot does not exist at {}", _snapshot_dir.native()));
+namespace {
+[[maybe_unused]] std::vector<std::tuple<size_t, std::vector<sstring>>> greedy_partitioning(std::vector<std::tuple<size_t, sstring>>&& files,
+                                                                                           size_t partitions) {
+    std::vector<std::tuple<size_t, std::vector<sstring>>> result(partitions);
+    std::ranges::sort(files, [](const auto& lhs, const auto& rhs) { return std::get<0>(lhs) > std::get<0>(rhs); });
+    for (auto&& [size, file] : files) {
+        auto& smallest = *std::ranges::min_element(result, [](const auto& lhs, const auto& rhs) { return std::get<0>(lhs) < std::get<0>(rhs); });
+        std::get<1>(smallest).emplace_back(std::move(file));
```

Comment:
The code is very opaque with the tuple and all the autos. Keep in mind that others will have to read it.